Sources/DocCCommon/FixedSizeBitSet.swift

```diff
@@ -0,0 +1,294 @@
+        // This is the index (bit offset) to the smallest value in the bit set.
+        Index(bit: UInt8(storage.trailingZeroBitCount))
+    }
+    
+    @inlinable
+    package var endIndex: Index {
+        // For a valid collection, the end index is required to be _exactly_ one past the last in-bounds index, meaning; `index(after: LAST_IN-BOUNDS_INDEX)`
+        // If the collection implementation doesn't satisfy this requirement, it will have an infinitely long `indices` collection.
+        // This either results in infinite implementations or hits internal preconditions in other Swift types that that collection has more elements than its `count`.
+        
+        // See `index(after:)` below for explanation of how the index after is calculated.
+        let lastInBoundsBit = UInt8(Storage.bitWidth &- storage.leadingZeroBitCount)
+        return Index(bit: lastInBoundsBit &+ UInt8((storage &>> lastInBoundsBit).trailingZeroBitCount))
+    }
+    
+    @inlinable
+    package func index(after currentIndex: Index) -> Index {
+        // To advance the index we have to find the next 1 bit _after_ the current bit.
+        // For example, consider the following 16 bits, where values are represented from left to right:
+        //   0110 0010 0110 0010
+        //
+        // To go from the first index to the second index, we need to count the number of 0 bits between it and the next 1 bit.
+        // We get this value by shifting the bits by one past the current index:
+        //   0110 0010 0110 0010
+        //                    ╰╴current index
+        //   0001 1000 1001 1000
+        //                   ~~~ 3 trailing zero bits
+        //
+        // The second index's absolute value is the one past the first index's value plus the number of trailing zero bits in the shifted value.
+        //
+        // For the third index we repeat the same process, starting by shifting the bits by one past second index:
+        //   0110 0010 0110 0010
+        //               ╰╴current index
+        //   0000 0001 1000 1001
+        //                      0 trailing zero bits
+        //
+        // This time there are no trailing zero bits in the shifted value, so the third index's absolute value is just one past the second index.
+        let shift = currentIndex.bit &+ 1
+        return Index(bit: shift &+ UInt8((storage &>> shift).trailingZeroBitCount))
+    }
+    
+    @inlinable
+    package func formIndex(after index: inout Index)  {
+        // See `index(after:)` above for explanation.
+        index.bit &+= 1
+        index.bit &+= UInt8((storage &>> index.bit).trailingZeroBitCount)
+    }
+    
+    @inlinable
+    package func distance(from start: Index, to end: Index) -> Int {
+        // To compute the distance between two indices we have to find the number 1 bit from the start index to (but excluding) the end index.
+        // For example, consider the following 16 bits, where values are represented from left to right:
```

Comment:
this and the other comment (above) should say "right to left"
```suggestion
        // For example, consider the following 16 bits, where values are represented from right to left:
```